src/main.rs

```diff
@@ -291,6 +312,13 @@ fn main_send(
         ))
         .expect("Listener should not exit before the sender.");
 
+    let rl = Arc::new(Mutex::new(Option::<RateLimiter>::None));
+
+    if let Some(mbps) = max_bandwidth_mbps {
+        let _rl = RateLimiter::new(mbps, Instant::now());
+        _ = rl.lock().unwrap().insert(_rl);
```

Comment:
Is the underscore to avoid shadowing the the `rl` Arc/Mutex? A prefix underscore in Rust usually means that a variable is intentionally not used, but that is not the case here, I would rename the outer `rl` to `rl_mutex` or something. Or even `limiter_mutex`, why leave the reader guessing at what RL means if we can just spell it out?

Reply:
`Option::insert` returns the current value in the option, which I don't need -- and is always `None`
I've renamed the `rl` variables, it's much clearer now